// vuepress-theme-hope/lib/node/handleThemeData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleThemeData = void 0;
const vuepress_shared_1 = require("@mr-hope/vuepress-shared");
const encrypt_1 = require("./encrypt");
const locales_1 = require("./locales");
const defaultRootOptions = {
    blog: {
        articleInfo: [
            "Author",
            "Original",
            "Date",
            "Category",
            "Tag",
            "ReadingTime",
        ],
        articlePerPage: 10,
        sidebarDisplay: "mobile",
    },
    encrypt: {
        global: false,
    },
    // layouts
    repoDisplay: true,
    navbarIcon: true,
    navbarAutoHide: "mobile",
    hideSiteNameonMobile: true,
    sidebar: "auto",
    sidebarIcon: true,
    headingDepth: 2,
    // appearance
    pure: false,
    iconPrefix: "",
    darkmode: "auto-switch",
    themeColor: {
        red: "#e74c3c",
        blue: "#3498db",
        green: "#3eaf7c",
        orange: "#f39c12",
        purple: "#8e44ad",
    },
    fullScreen: true,
};
/**
 * Assign default options to `themeConfig`
 */
const handleThemeData = (app, themeOptions) => {
    // ensure locales
    if (!themeOptions.locales)
        themeOptions.locales = {};
    // ensure root locales
    if (!themeOptions.locales["/"])
        themeOptions.locales["/"] = {};
    // merge default options
    (0, vuepress_shared_1.deepAssignReverse)(defaultRootOptions, themeOptions);
    const themeData = themeOptions;
    // handle encrypt options
    (0, encrypt_1.resolveEncrypt)(themeData.encrypt);
    // assign locale data to `themeConfig`
    themeOptions.locales = (0, vuepress_shared_1.getLocales)(app, locales_1.themeLocalesData, themeOptions.locales);
    if (app.env.isDebug)
        console.log("Theme config: ", themeOptions);
};
exports.handleThemeData = handleThemeData;
