// vuepress-theme-hope/lib/node/theme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.themeHope = void 0;
const utils_1 = require("@vuepress/utils");
const alias_1 = require("./alias");
const bundler_1 = require("./bundler");
const handleThemeData_1 = require("./handleThemeData");
const extends_1 = require("./extends");
const layout_1 = require("./layout");
const plugins_1 = require("./plugins");
const socialMedia_1 = require("./socialMedia");
const themeColor_1 = require("./themeColor");
const themeHope = ({ plugins = {}, ...themeOptions }, app) => {
    const enableBlog = Boolean(plugins.blog);
    (0, handleThemeData_1.handleThemeData)(app, themeOptions);
    (0, plugins_1.usePlugin)(app, plugins);
    if (enableBlog) {
        const icons = (0, socialMedia_1.checkSocialMediaIcons)(themeOptions);
        void app.writeTemp(`theme-hope/socialMedia.js`, `export const icons = ${JSON.stringify(icons)}`);
    }
    return {
        name: "vuepress-theme-hope",
        alias: (0, alias_1.getAlias)(app),
        define: () => ({
            ENABLE_BLOG: enableBlog,
        }),
        extendsPage: (page) => (0, extends_1.extendsPage)(themeOptions, plugins, page, app.env.isDev),
        onInitialized: (app) => (0, bundler_1.updateBundlerOptions)(app),
        onPrepared: () => (0, themeColor_1.writeThemeColorScss)(app, themeOptions),
        plugins: (0, plugins_1.getPluginConfig)(app, plugins, themeOptions),
        layouts: (0, layout_1.getLayoutConfig)(app, plugins),
        templateBuild: utils_1.path.resolve(__dirname, "../../templates/index.build.html"),
        clientAppEnhanceFiles: [
            utils_1.path.resolve(__dirname, "../client/appEnhance.js"),
            utils_1.path.resolve(__dirname, "../client/module/navbar/appEnhance.js"),
            utils_1.path.resolve(__dirname, "../client/module/sidebar/appEnhance.js"),
            ...(enableBlog
                ? [utils_1.path.resolve(__dirname, "../client/module/blog/appEnhance.js")]
                : []),
        ],
        clientAppSetupFiles: [
            ...(enableBlog
                ? [utils_1.path.resolve(__dirname, "../client/module/blog/appSetup.js")]
                : []),
            utils_1.path.resolve(__dirname, "../client/module/outlook/appSetup.js"),
            utils_1.path.resolve(__dirname, "../client/module/sidebar/appSetup.js"),
        ],
    };
};
exports.themeHope = themeHope;
