// vuepress-theme-hope/lib/client/composables/pageInfo.js
import { getAuthor, getCategory, getDate, getTag, } from "@mr-hope/vuepress-shared/lib/client";
import { usePageData, usePageFrontmatter, usePageLang } from "@vuepress/client";
import { computed, inject, reactive } from "vue";
import { usePure, useThemeData, useThemeLocaleData } from "./themeData";
export const usePageAuthor = () => {
    const themeData = useThemeData();
    const frontmatter = usePageFrontmatter();
    return computed(() => {
        const { author } = frontmatter.value;
        if (author)
            return getAuthor(author);
        if (author === false)
            return [];
        return getAuthor(themeData.value.author, false);
    });
};
export const usePageCategory = () => {
    const frontmatter = usePageFrontmatter();
    return computed(() => getCategory(frontmatter.value.category).map((name) => ({
        name,
        // this is a hack
        path: ENABLE_BLOG
            ? inject(Symbol.for("categoryMap"))?.value.map[name]
                ?.path || ""
            : "",
    })));
};
export const usePageTag = () => {
    const frontmatter = usePageFrontmatter();
    return computed(() => getTag(frontmatter.value.tag).map((name) => ({
        name,
        // this is a hack
        path: ENABLE_BLOG
            ? inject(Symbol.for("tagMap"))?.value.map[name]?.path ||
                ""
            : "",
    })));
};
export const usePageDate = () => {
    const frontmatter = usePageFrontmatter();
    const page = usePageData();
    const pageLang = usePageLang();
    return computed(() => {
        const { date } = frontmatter.value;
        const options = { lang: pageLang.value, type: "date" };
        if (date)
            return getDate(date, options);
        const { createdTime } = page.value.git || {};
        if (createdTime)
            return getDate(new Date(createdTime), options);
        return null;
    });
};
export const usePageInfo = () => {
    const themeLocale = useThemeLocaleData();
    const page = usePageData();
    const frontmatter = usePageFrontmatter();
    const author = usePageAuthor();
    const category = usePageCategory();
    const tag = usePageTag();
    const date = usePageDate();
    const pure = usePure();
    return reactive({
        config: frontmatter.value.pageInfo === false
            ? false
            : frontmatter.value.pageInfo || themeLocale.value.pageInfo,
        author: author.value,
        category: category.value,
        date: date.value,
        tag: tag.value,
        isOriginal: frontmatter.value.isOriginal,
        readingTime: page.value.readingTime,
        visitor: frontmatter.value.visitor !== false,
        color: !pure.value,
        hint: !pure.value,
    });
};
